modules/minimint-mint/src/tiered/coins.rs

```diff
@@ -84,29 +84,55 @@ impl<C> Coins<C> {
         }
     }
 }
+
+/// Select coins with total amount of *at least* `amount`. If more than requested amount of coins
+/// are returned it was because exact change couldn't be made.
+/// The caller can request change from the federation.
 impl<C> Coins<C>
 where
     C: Clone,
 {
-    pub fn select_coins(&self, mut amount: Amount) -> Option<Coins<C>> {
+    pub fn select_coins(&self, amount: Amount) -> Option<Coins<C>> {
```

Comment:
This is really the main change in the PR: `Coins::select_coins` may return more than requested amount of coins. Everything else is really making room for this change.